File: lab2/lab_python_oop/square.py
from lab_python_oop.rectangle import Rectangle

class Square(Rectangle):
    figure_type = "Квадрат"
    def __init__(self, width, color):
        super().__init__(width, width, color)
    @classmethod
    def get_type(cls):
        return cls.figure_type
    def __repr__(self):
       return 'Тип: {}, Сторона: {}, Цвет: {}, Площадь: {}'.format(Square.get_type(), self.width, self.color.color, super().get_area())

File: lab3_rust/geometrics/figures/src/square.rs
use std::fmt;
use crate::figure::Figure;
use crate::color::Color;
use crate::rect::Rect;

pub struct Square{
    figure_type: String,
    rect: Rect
}

impl Square{
    pub fn new(s: f64, col: &Color) -> Square{
        let c: Color = Color::new(col.get_color());
        let r: Rect = Rect::new(s, s, &c);
        Square {figure_type: "square".to_string(), rect: r}
    }
    pub fn get_type(&self) -> String{
        self.figure_type.clone()
    }
}

impl Figure for Square{
    fn get_area(&self) -> f64{
        self.rect.get_area()
    }
    fn repr(&self) -> String{
        format!("Тип: {}, Сторона: {}, Цвет: {}, Площадь: {}", self.get_type(), self.rect.width, self.rect.color.get_color(), self.get_area())
    }
}

impl std::fmt::Display for Square{
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        write!(f, "{}", self.repr())
    }
}